protocols/kad/src/behaviour.rs

```diff
@@ -667,6 +667,17 @@ where
         self.queries.add_iter_closest(target.clone(), peers, inner)
     }
 
+    /// Returns closest peers to the given key; takes peers from local routing table
+    pub fn local_closest_peers<'s, 'k: 's, K: 's>(
+        &'s mut self,
+        key: &'k kbucket::Key<K>,
```

Comment:
```suggestion
        key: K,
        // ...
        where
          K: Into<kbucket::Key<K>> + Into<Vec<u8>> + Clone,
```

Why not mirror the signature of `get_closest_peers`?

Reply:
Because we have to pass `key` as a reference, so function can return an `impl Iterator` that would use that reference.

We can't pass key by value, because then `fn local_closest_peers` will own it and drop, thus invalidating `impl Iterator`. Borrow checker would be unhappy about that :)

Reply:
It's possible to do it this way:
```
    pub fn local_closest_peers<'s, 'k: 's, K: 's>(
        &'s mut self,
        key: &'k K,
    ) -> impl Iterator<Item = kbucket::Key<PeerId>> + 's
    where
        K: AsRef<kbucket::Key<K>> + Clone,
    {
        self.kbuckets.closest_keys(key.as_ref())
    }
```
Do you find that better than the current version?

P.S. `Into<Vec<u8>>` isn't required for `get_local_closest_peers`

Reply:
On a second thought, `K: AsRef<kbucket::Key<K>> + Clone` doesn't make sense. It does make sense to do `K: Into<Key<K>>`, cuz then if refers to the following `From` impls:
```
impl From<Multihash> for Key<Multihash>
impl From<PeerId> for Key<PeerId>
impl From<Vec<u8>> for Key<Vec<u8>>
```

But since there aren't any analogous `AsRef<Key<_>>` impls, the bound `K: AsRef<kbucket::Key<K>>` doesn't make sense.

Reply:
So I see best to stay with `key: &'k kbucket::Key<K>` and rename `K` to `T` for naming consistency. WDYT?